src/Mvc/Mvc.Razor.RuntimeCompilation/src/DependencyInjection/RazorRuntimeCompilationMvcCoreBuilderExtensions.cs

```diff
@@ -95,6 +96,7 @@ internal static void AddServices(IServiceCollection services)
                     // TagHelperDescriptorProviders (actually do tag helper discovery)
                     builder.Features.Add(new DefaultTagHelperDescriptorProvider());
                     builder.Features.Add(new ViewComponentTagHelperDescriptorProvider());
+                    builder.SetCSharpLanguageVersion(csharpCompiler.ParseOptions.LanguageVersion);
```

Comment:
As long as the parse options are updated early enough when everything is spinning up 👍